### PR TITLE
OCPBUGSM-17301 Node fails to boot due to tailing enter in the SSH key

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -937,9 +940,11 @@ func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer
 	}
 
 	if sshPublicKey := swag.StringValue(params.ClusterUpdateParams.SSHPublicKey); sshPublicKey != "" {
+		sshPublicKey := strings.TrimSpace(sshPublicKey)
```
Comment (filanov) [inline internal/bminventory/inventory.go:943]: Maybe you can validate that there are no spaces in the middle.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -365,9 +365,11 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 	}
 
 	if sshPublicKey := swag.StringValue(&cluster.SSHPublicKey); sshPublicKey != "" {
+		sshPublicKey = strings.TrimSpace(cluster.SSHPublicKey)
```
Comment (RazRegev) [inline internal/bminventory/inventory.go:368]: maybe we should move this line outside the if, so we will cover also cases of key=="\n\n"

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (tsorya) [reply internal/bminventory/inventory.go]: i can add it to validation, i think will do it in another PR

File: internal/bminventory/inventory.go (hunk 2 above)
Comment (tsorya) [reply internal/bminventory/inventory.go:368]: it covers it. I added to if cause if validates that SSHPublicKey is set at all

**Review by RazRegev — APPROVED** [review]

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (filanov) [reply internal/bminventory/inventory.go]: Please open a ticket for it

Comment (tsorya) [reply internal/bminventory/inventory.go]: tested it. Validation will fail if there is \n in the middle

Comment (RazRegev) [reply internal/bminventory/inventory.go]: @filanov this cases are already covered (checked)

**Pull request merged**